proto/iotics/api/input.proto

```diff
@@ -72,6 +79,84 @@ message DeleteInputResponse {
 }
 // ---------------------------------------
 
+// CreateInputRequest describes a created input.
+message CreateInputRequest {
+  // CreateInputRequest payload.
+  message Payload {
+    // Input identifier string representation (simple string) (unique within the scope of a twin identifier's input set)
+    string id = 1;
+  }
+
+  // Arguments describes the mandatory arguments to identify the twin owning this input
+  message Arguments {
+    // Identifier of the twin owning this input
+    TwinID twinId = 1;
+  }
+
+  // CreateInputResponse headers
+  Headers headers = 1;
+  // CreateInputRequest arguments
+  Arguments args = 2;
+  // CreateInputResponse payload
+  Payload payload = 3;
+}
+// ---------------------------------------
+
+// CreateInputResponse describes a created input.
+message CreateInputResponse {
+  // CreateInputResponse payload.
+  message Payload {
+    // Created input
+    InputID inputId = 1;
+  }
+
+  // CreateInputResponse headers
+  Headers headers = 1;
+  // CreateInputResponse payload
+  Payload payload = 2;
+}
+// ---------------------------------------
+
+// UpdateInputRequest is used to update an input.
+message UpdateInputRequest {
+  // UdpateInputRequest payload used to update one or more field
+  message Payload {
+    // StoreLast dictates if we should store the last received message for this input
+    google.protobuf.BoolValue storeLast = 1;
```

Comment:
Inputs don't have the exact same fields as Feeds - take a look at the twin protos/examples docs/metadata repos to see how inputs are constructed.

Reply:
In fact, just look at the the `UpsertInputWithMeta` above :)